amgut/lib/data_access/ag_data_access.py

```diff
@@ -386,6 +386,68 @@ def updateAGBarcode(self, barcode, ag_kit_id, site_sampled,
                                     refunded, withdrawn])
         self.connection.commit()
 
+    def registerHandoutKit(self, ag_login_id, supplied_kit_id):
+        """
+        Returns
+        -------
+        int
+            1:  success
+            -1: insert failed due to IntegrityError
+
+        Notes
+        -----
+        Whatever is passed as kit_password will be added AS IS. This means you
+        must hash the password before passing, if desired.
+        """
+        kitinfo = self.getAGHandoutKitDetails(supplied_kit_id)
+        printresults = self.checkPrintResults(supplied_kit_id)
+        if printresults is None:
+            printresults = 'n'
+
+        conn_handler = SQLConnectionHandler()
+        queue = "register_%s" % supplied_kit_id
+        conn_handler.create_queue(queue)
+        # insert the actual kit into the kit table.
+        # Needs to be done directly because ag_kit_id needed for rest
+        kit_id = conn_handler.execute_fetchone(
+            """INSERT INTO ag_kit
+            (ag_login_id, supplied_kit_id, kit_password, swabs_per_kit,
+            kit_verification_code, print_results)
+            VALUES (%s, %s, %s, %s, %s, %s) RETURNING ag_kit_id""",
+            (ag_login_id, supplied_kit_id, kitinfo['password'],
+             kitinfo['swabs_per_kit'], kitinfo['verification_code'],
+             printresults))[0]
+
+        # get the barcodes and add them to the queue
+        sql = "SELECT barcode FROM ag_handout_kits WHERE kit_id = %s"
+        for row in conn_handler.execute_fetchall(sql, [supplied_kit_id]):
+            barcode = row[0]
+            conn_handler.add_to_queue(
+                queue, "INSERT INTO barcode (barcode) VALUES (%s)", [barcode])
+            conn_handler.add_to_queue(
+                queue, "INSERT INTO project_barcode (project_id, barcode) "
+                "VALUES (1, %s)", [barcode])
```

Comment:
Looking at the schema, it looks like kits aren't tied to projects, but rather individual barcodes. So this might be necessary, unless you're saying that all barcodes in a kit are tied to the same project automatically upon the kit's creation (but I'm not seeing where that is done in the code).